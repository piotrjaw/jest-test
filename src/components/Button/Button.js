import React from 'react';
import PropTypes from 'prop-types';

const Button = () => (
  <button />
);

Button.propTypes = {
};

export default Button;
