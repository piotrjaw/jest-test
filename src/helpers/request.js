import axios from 'axios';

const contructUrl = (url) => `http://localhost:5000${url}`;

export const del = (url) => axios.delete(contructUrl(url));

export const get = (url) => axios.get(contructUrl(url));

export const post = (url, data) => axios.post(contructUrl(url), data);

export const put = (url, data) => axios.put(contructUrl(url), data);
