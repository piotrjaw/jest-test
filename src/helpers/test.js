import React from 'react';

export const getWrapper = ({ Component, method, props }) => {
  return method(
    <Component
      { ...props }
    >
      { props.children }
    </Component>
  );
};

export const getMockStore = (mockFn, state = {}) => ({
  subscribe: mockFn(),
  dispatch: mockFn(),
  getState: mockFn(() => ({
    ...state,
  })),
});
