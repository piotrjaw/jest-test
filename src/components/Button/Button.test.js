import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { getWrapper } from '../../helpers/test';
import Button from './Button';

const setupWrapper = (config = {}) => {
  const { method = shallow, props = {} } = config;
  return getWrapper({
    Component: Button,
    method,
    props: {
      ...props
    }
  });
};

let wrapper;

beforeEach(() => {
  wrapper = setupWrapper();
});

afterEach(() => {
  expect(toJSON(wrapper.render())).toMatchSnapshot();
  wrapper = null;
});
