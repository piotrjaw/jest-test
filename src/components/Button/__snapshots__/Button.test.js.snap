// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should call onClick when clicked 1`] = `
<button
  type="button"
/>
`;

exports[`Button component should render 1`] = `
<button
  type="button"
/>
`;

exports[`Button component should render HTML children 1`] = `
<button
  type="button"
>
  <span>
    Test
  </span>
</button>
`;

exports[`Button component should render text children 1`] = `
<button
  type="button"
>
  Test
</button>
`;
